File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    score = {
        k:v for k,v in zip(name,yearning)
    }
    for p in photo:
        num = 0
        for pp in p:
            if pp in name:
                num = num + score[pp]
        answer.append(num)
    return answer

File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photoAlbum) {
    vector<int> answer;
    map<string, int> missScore;
    for(int i=0;i<name.size();i++){
        missScore[name[i]] = yearning[i];
    }
    for(vector<string> photos : photoAlbum){
        int score = 0;
        for(string photo : photos){
            score += missScore[photo];
        }
        answer.push_back(score);
    }
    return answer;
}